tests/product/timing_test_decorator.py

```diff
@@ -0,0 +1,65 @@
+import logging
```

Comment:
@kokosing this decorator is meant to be complementary to your commit. It can used to annotate some key functions and measure their runtime so that we can quickly identify which function causes a runtime increase when that happens.

However, I think we should get rid of the log files and instead print times on stderr or stdout. That would greatly simplify the code. Also we could not have any methods decorated by default and instead have a PyDoc section on the decorator telling future developers what methods should be annotated if they wish to debug timing problems.